qcengine/programs/molpro.py

```diff
@@ -35,17 +35,23 @@ def found(self, raise_error: bool=False) -> bool:
     def get_version(self) -> str:
         self.found(raise_error=True)
 
-        #which_prog = which('molpro')
-        #if which_prog not in self.version_cache:
-        #    success, output = execute([which_prog, "v.inp"], {"v.inp": ""})
-
-        #    if success:
-        #        for line in output["stdout"].splitlines():
-        #            if 'GAMESS VERSION' in line:
-        #                branch = ' '.join(line.strip(' *\t').split()[3:])
-        #        self.version_cache[which_prog] = safe_version(branch)
-
-        #return self.version_cache[which_prog]
+        name_space = {'molpro_uri': 'http://www.molpro.net/schema/molpro-output'}
+        which_prog = which('molpro')
+        if which_prog not in self.version_cache:
+            success, output = execute([which_prog, "version.inp", "-d", ".", "-W", "."],
+                                      infiles={"version.inp": ""},
+                                      outfiles=["version.out", "version.xml"])
+
+            if success:
+                tree = ET.ElementTree(ET.fromstring(output["outfiles"]["version.xml"]))
+                root = tree.getroot()
+                version_tree = root.find('molpro_uri:job/molpro_uri:platform/molpro_uri:version', name_space)
+                year = version_tree.attrib['major']
+                minor = version_tree.attrib['minor']
+                molpro_version = year + "." + minor
+                self.version_cache[which_prog] = safe_version(molpro_version)
```

Comment:
We need an else here that raises an exception. Usually this is a `TypeError`.

Reply:
Sounds good! I'm a little confused about how this is treated as a `TypeError`. Could you give me an example of how the error message would read?

Reply:
Away from a computer at the moment. Can you look at the psi4 example?

Reply:
Taking a look at the psi4 example the `TypeError` is for catching custom versions of the Psi4 program? 

```
candidate_version = self.version_cache[which_prog]
if "undef" in candidate_version:
    raise TypeError("Using custom build without tags. Please pull git tags with `git pull origin master --tags`.")
```

I don't know if this exactly relevant for Molpro since I believe no matter what you compile it will always have a version enabled. Also I don't see any other examples for any of the other backends so I'm not entirely sure what I should be catching here.

Reply:
Right, so this would be similar here:
```python
if version < (2018, 1):
    raise TypeError("QCEngine requires Molpro version 2018.1 or great")
```

Reply:
Yeah, I'd say merge as is and only raise error if there's a problem with version computing (as with psi and the tags). Version too low belongs to [`is_program_new_enough`](https://github.com/MolSSI/QCEngine/blob/master/qcengine/testing.py#L35) imo.